src/middlewared/middlewared/plugins/account.py

```diff
@@ -873,6 +874,9 @@ def do_delete(self, audit_callback, pk, options):
         user = self.middleware.call_sync('user.get_instance', pk)
         audit_callback(user['username'])
 
+        if user['username'] == self.middleware.call_sync('auth.me', app=app)['pw_name']:
```

Comment:
You should be able to extract current user token from `app` without calling auth.me.

Reply:
Working on a test case now, will get back to this